lambdas/create_appointment.py

```diff
@@ -1,45 +1,113 @@
 # Written by Maeve Newman
-# 4/08/2020
+# Updated 4/22/2020
 
 import json
-import boto3
-#from package import db_config
+from package.query_db import query
+from package.lambda_exception import LambdaException
+import time
+import datetime
 
 # function puts the appointment details in the database
-# Input: supporter_id, time, type, duration, location
+# Inputs: supporter_id, time_of_appt, type, duration, method, location
 # Output: 201 Created
 def lambda_handler(event, context):
-    appointment_id = event['appointment_id']
-    supporter_id = event['supporter_id']
-    time = event['time_of_appt']
+    # take in lambda input
+    student = int(event['student_id'])
+    supporter = int(event['supporter_id'])
+    time_of_appt = event['time_of_appt']
     appt_type = event['type']
-    duration = event['duration']
-    location = event['location']
+    duration = int(event['duration'])
     method = event['method']
+    location = event['location']
+    
+    if 'comment' in event:
+        comment = event['comment']
+    else:
+        comment = ""
+    
+    # check that student is in DB
+    sql = "SELECT student_id FROM students WHERE student_id = :student"
+    sql_parameters = [
+        {'name' : 'student', 'value': {'longValue': student}}
+    ]
+    check_student = query(sql, sql_parameters)
 
-    # connect to db
-    client = boto3.client('rds-data')
+    # if student does not exist in DB, return error
+    if(check_student['records'] == []):
+        return{
+            'body': json.dumps("Student not found."),
+            'statusCode': 404
+        }
+
+    # check that supporter is in DB
+    sql = "SELECT supporter_id FROM supporters WHERE supporter_id = :supporter"
+    sql_parameters = [
+        {'name' : 'supporter', 'value': {'longValue': supporter}}
+    ]
+    check_supporter = query(sql, sql_parameters)
+    
+    # if supporter does not exist in DB, return error
+    if(check_supporter['records'] == []):
+        return{
+            'body': json.dumps("Supporter not found."),
+            'statusCode': 404
+        }
+    
+    # if supporter is in DB, set id variable for query
+    student_id = student
+    supporter_id = supporter
```

Comment:
Not a big deal but I would initialize in the event to student_id and supporter_id rather than changing it when checking if they exist. Otherwise looks good overall